File: webapp/src/Websocket/types/EnergyPricingSettings.ts
export type EnergyPricingSettingsWrite = {
    service: string;
    neutralSpan?: number;
    avgMultiplier?: number;
}

export type EnergyPricingSettingsRead = {
    serviceConfigs: EnergyPriceConfig[],
}

export type EnergyPriceConfig = {
    service: string;
    neutralSpan: number;
    avgMultiplier: number;
    categorizedPrices: CategorizedPrice[];
    priceDecision?: PriceDecision;
}

export type CategorizedPrice = {
    category: PriceCategory;
    price: Price;
}

export type PriceCategory = 'cheap' | 'neutral' | 'expensive';

export type Price = {
    from: string;
    to: string;
    price: number;
}

export type PriceDecision = {
    current: PriceCategory;
    changesAt: string;
    changesInto: PriceCategory;
}

File: src/main/kotlin/com/dehnes/smarthome/api/dtos/EnergyPricingSettings.kt
package com.dehnes.smarthome.api.dtos

import com.dehnes.smarthome.energy_pricing.EnergyPriceConfig

data class EnergyPricingSettingsRead(
    val serviceConfigs: List<EnergyPriceConfig>,
)

data class EnergyPricingSettingsWrite(
    val service: String,
    val neutralSpan: Double?,
    val avgMultiplier: Double?,
)
